src/scripts/user_script_hero_vo.lua

```diff
@@ -0,0 +1,244 @@
+				cis_hero_darthmaul = "cis_hero_maul_",
+				cis_hero_countdooku = "cis_hero_dooku_",
+				cis_hero_jangofett = "cis_hero_jango_",
+			},
+			Them =
+			{
+				rep_hero_yoda = "cis_hero_yoda_",
+				rep_hero_macewindu = "cis_hero_windu_",
+				rep_hero_anakin = "cis_hero_anakin_",
+				rep_hero_aalya = "cis_hero_aayla_",
+				rep_hero_kiyadimundi = "cis_hero_mundi_",
+				rep_hero_obiwan = "cis_hero_kenobi_",
+			},
+		},
+	},
+	otherTeam = {},
+	heroClassName = {},
+	heroClassIndex = {},
+	hasStarted = false,
+}
+
+function HeroVOScript:New(o)
+	o = o or {}
+	setmetatable(o, self)
+	self.__index = self
+	return o
+end
+
+function HeroVOScript:Start()
+	if self.hasStarted then return end
+	
+	local getTeamName = function(teamPtr)
+		local teamName = nil
+		
+		if GetTeamFactionId(teamPtr) == 1 then
+			teamName = "all"
+		elseif GetTeamFactionId(teamPtr) == 2 then
+			teamName = "imp"
+		elseif GetTeamFactionId(teamPtr) == 3 then
+			teamName = "rep"
+		elseif GetTeamFactionId(teamPtr) == 4 then
+			teamName = "cis"
+		end
+		
+		return teamName
+	end
+	
+	local initHeroVO = function(teamPtr)
+		OnCharacterDeathTeam(
+			function(character, killer)
+				local charClass = GetCharacterClass(character)
+				if charClass == self.heroClassIndex[teamPtr] then
+					local usVO = self.heroVOList[GetTeamFactionId(teamPtr)].Us[self.heroClassName[teamPtr]]
+					local themVO = self.heroVOList[GetTeamFactionId(self.otherTeam[teamPtr])].Them[self.heroClassName[teamPtr]]
+					
+					if usVO then
+						BroadcastVoiceOver(usVO .. "exit", teamPtr)
+					end
+					if themVO then
+						BroadcastVoiceOver(themVO .. "exit", self.otherTeam[teamPtr])
+					end
+				end
+			end,
+			teamPtr
+		)
+		
+		OnCharacterSpawnTeam(
+			function(character)
+				local charClass = GetCharacterClass(character)
+				local heroIndex = self.heroClassIndex[teamPtr]
+				
+				if charClass == heroIndex then
+					local usVO = self.heroVOList[GetTeamFactionId(teamPtr)].Us[self.heroClassName[teamPtr]]
+					local themVO = self.heroVOList[GetTeamFactionId(self.otherTeam[teamPtr])].Them[self.heroClassName[teamPtr]]
+					
+					if usVO then
+						BroadcastVoiceOver(usVO .. "enter", teamPtr)
+					else
+						if getTeamName(teamPtr) then
+							BroadcastVoiceOver("team_bonus_leader_us_" .. getTeamName(teamPtr), teamPtr)
+						end
+					end
+					if themVO then
+						BroadcastVoiceOver(themVO .. "enter", self.otherTeam[teamPtr])
+					else
+						if getTeamName(self.otherTeam[teamPtr]) then
+							BroadcastVoiceOver("team_bonus_leader_them_" .. getTeamName(self.otherTeam[teamPtr]), self.otherTeam[teamPtr])
+						end
+					end
+				end
+			end,
+			teamPtr
+		)
+	end
+	
+	for i = 1, self.numTeams do
+		if self.heroClassName[i] then
+			self.heroClassIndex[i] = (GetTeamClassCount(i) - 1)
+			self.otherTeam[i] = (i == 2 and 1 or 2)
+			initHeroVO(i)
+		end
+	end
+	
+	self.hasStarted = true
+end
+
+local HeroVOClasses = {}
+local HeroVONumTeams = 0
```

Comment:
In the past, I've had issues with local variables in the top-most scope of a script. I can't remember what the issues were, but I think there were significant differences in how they were processed by the Steam and retail versions of the game.

Reply:
I wasn't aware there was a difference. I'll look into it. What would you recommend I change exactly? I only have the Steam versions of classic and CC, so I can't test retail.

Reply:
To be honest, changing them to global variable, and prefixing them with a g (like `gHeroVOClasses`) should do the trick, unless there can be multiple instances of this script (unlikely).

Reply:
wouldn't it be better to have them as part of the HeroVOScript table/object/namespace